message_ix/model/MESSAGE/model_core.gms

```diff
@@ -572,6 +584,47 @@ $macro COMMODITY_BALANCE(node,commodity,level,year,time) (
         - input(location,tec,vintage,year,mode,node,commodity,level,time2,time)                                        \
         * duration_time_rel(time,time2) * ACT(location,tec,vintage,year,mode,time2) )                                  \
+* +++++
+* commodity input and output associated with construction of new technology capacity (during vintage period)
+  + SUM( (location,tec)$( inv_tec(tec) AND map_tec(location,tec,year) ),                                               \
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time'
+        output_cap_new(location,tec,year,node,commodity,level,time)                                                    \
+        * CAP_NEW(location,tec,year)                                                                                   \
+* input by all new capacity of technologies located at 'location' taking from 'node' and 'time'
+        - input_cap_new(location,tec,year,node,commodity,level,time)                                                   \
+        * CAP_NEW(location,tec,year) )                                                                                 \
+* commodity input and output associated with retirement of technology capacity (via differentials of capacity of successive periods)
+* for first model period (differential with historical remaining capacity)
+  + SUM( (location,tec,vintage,year2)$( inv_tec(tec) AND map_tec_lifetime(location,tec,vintage,year2)                  \
+            AND first_period(year) AND seq_period(year2,year) ),                                                       \
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time' distributed over years of periods
+        output_cap_ret(location,tec,vintage,node,commodity,level,time)                                                 \
+        * ( remaining_capacity(node,tec,vintage,year2) - CAP(location,tec,vintage,year) )                              \
+        / duration_period(year)                                                                                        \
+* input by all new capacity of technologies located at 'location' taking from 'node' and 'time' distributed over years of periods
+        - input_cap_ret(location,tec,vintage,node,commodity,level,time)                                                \
+        * ( remaining_capacity(node,tec,vintage,year2) - CAP(location,tec,vintage,year) )                              \
+        / duration_period(year) )                                                                                      \
+* for other model periods (differential with installed capacity of preceding period)
+  + SUM( (location,tec,vintage,year2)$( inv_tec(tec) AND map_tec_lifetime(location,tec,vintage,year2)                  \
+            AND NOT first_period(year) AND seq_period(year2,year) ),                                                   \
```

Comment:
Another suspected issue here is:
For the capacity ending its activity in e.g. `year2=750`, this part of the equation brings in the term `CAP(location,tec,vintage,year)` where `year=760`. And it appears that while solving the LP model, a spurious non-zero value is assigned to this CAP of `year=760`, which is after the capacity is retired.